sdk/go/common/resource/properties_path_test.go

```diff
@@ -733,6 +733,75 @@ func TestReset(t *testing.T) {
 			})},
 			nil,
 		},
+		{
+			"Nested path, secret old",
+			PropertyPath{"root", "secret"},
+			PropertyMap{"root": MakeSecret(NewProperty(PropertyMap{"secret": NewProperty(1.0)}))},
+			PropertyMap{"root": NewProperty(PropertyMap{"secret": NewProperty(2.0)})},
+			&PropertyMap{"root": NewProperty(PropertyMap{"secret": NewProperty(1.0)})},
```

Comment:
This behaviour seems a little risky to me.  We're replacing the non-secret "2.0" with a secret "1.0", but what we're getting back is a non-secret "1.0", which seems like we lost the secretness property of "1.0".  It feels like we should preserve that or is there something I'm missing why this is okay?  

Maybe we could make sure both old and new have the same properties before allowing the reset?

Reply:
Yeh I think you're right we need to keep the secretness here but I think at the level of the thing being replaced, not the whole object. So this should have:
```&PropertyMap{"root": NewProperty(PropertyMap{"secret": MakeSecret(NewProperty(1.0))})},```
Not a big difference in this case, but if there were other child properties we shouldn't force those to be secrets just because we've reset one of their siblings that used to be secret.